src/core/ticker/Ticker.js

```diff
@@ -193,55 +192,106 @@ export default class Ticker
     }
 
     /**
-     * Calls {@link module:eventemitter3.EventEmitter#on} internally for the
-     * internal 'tick' event. It checks if the emitter has listeners,
-     * and if so it requests a new animation frame at this point.
+     * Register a handler for tick events. Calls continuously unless
+     * it is removed or the ticker is stopped.
      *
      * @param {Function} fn - The listener function to be added for updates
      * @param {Function} [context] - The listener context
+     * @param {number} [priority=PIXI.UPDATE_PRIORITY.NORMAL] - The priority for emitting
      * @returns {PIXI.ticker.Ticker} This instance of a ticker
      */
-    add(fn, context)
+    add(fn, context, priority = UPDATE_PRIORITY.NORMAL)
     {
-        this._emitter.on(TICK, fn, context);
-
-        this._startIfPossible();
-
-        return this;
+        return this._addListener(new TickerListener(fn, context, priority));
     }
 
     /**
-     * Calls {@link module:eventemitter3.EventEmitter#once} internally for the
-     * internal 'tick' event. It checks if the emitter has listeners,
-     * and if so it requests a new animation frame at this point.
+     * Add a handler for the tick event which is only execute once.
      *
      * @param {Function} fn - The listener function to be added for one update
      * @param {Function} [context] - The listener context
+     * @param {number} [priority=PIXI.UPDATE_PRIORITY.NORMAL] - The priority for emitting
+     * @returns {PIXI.ticker.Ticker} This instance of a ticker
+     */
+    addOnce(fn, context, priority = UPDATE_PRIORITY.NORMAL)
+    {
+        return this._addListener(new TickerListener(fn, context, priority, true));
+    }
+
+    /**
+     * Internally adds the event handler so that it can be sorted by priority.
+     * Priority allows certain handler (user, AnimatedSprite, Interaction) to be run
+     * before the rendering.
+     *
+     * @private
+     * @param {TickerListener} listener - Current listener being added.
      * @returns {PIXI.ticker.Ticker} This instance of a ticker
      */
-    addOnce(fn, context)
+    _addListener(listener)
     {
-        this._emitter.once(TICK, fn, context);
+        // For attaching to head
+        let current = this._head.next;
+        let last = this._head;
```

Comment:
I'd prefer that `last` be named `previous` to prevent anyone from thinking that it represents the end of the list, but other than that this all looks good.